internal/schema/typeref.go

```diff
@@ -119,6 +107,65 @@ func (r *TypeRef) IsScalarID() bool {
 	return r.OfType != nil && r.OfType.Kind == KindScalar && r.GetTypeName() == "ID"
 }
 
+// IsNonNull walks down looking for NON_NULL kind, however that can appear
+// multiple times, so this is likely a bit deceiving...
+// Example:
+// {
+//    "name": "tags",
+//    "description": "An array of key-values pairs to represent a tag. For example:  Team:TeamName.",
+//    "type": {
+//     "kind": "NON_NULL",
+//     "ofType": {
+//      "kind": "LIST",
+//      "ofType": {
+//       "kind": "NON_NULL",
+//       "ofType": {
+//        "name": "TaggingTagInput",
+//        "kind": "INPUT_OBJECT"
+//       }
+//      }
+//     }
+//    }
+//   }
+//  ]
+// }
 func (r *TypeRef) IsNonNull() bool {
```

Comment:
Here too we probably only want to test the first kind.

Reply:
Non Null can appear multiple time, and wouldn't have to be the first.  Example:

`LIST, NON_NULL, Scalar`